lib/OpenID4VCIClient.ts

```diff
@@ -70,17 +71,80 @@ export class OpenID4VCIClient {
     return this._serverMetadata;
   }
 
-  public async acquireAccessToken({ pin, clientId }: { pin?: string; clientId?: string }): Promise<AccessTokenResponse> {
+  public createAuthorizationRequestUrl({ clientId, codeChallengeMethod, codeChallenge, redirectUri, scope }: AuthorizationRequestOpts): string {
+    if (!scope) {
+      throw Error('Please provide a scope. authorization_details based requests are not supported at this time');
+    }
+
+    if (!this._serverMetadata.openid4vci_metadata.authorization_endpoint) {
+      throw Error('Server metadata does not contain authorization endpoint');
+    }
+
+    // make sure the first scope is 'openid'
+    if (scope.includes('openid')) {
```

Comment:
See other remarks about this needing to be a space separated string. The order doesn't matter (see other comment)

Reply:
I was converting the array into a space separated string because I thought it was a more convinient API. That being said, staying close to the spec makes total sense. I've refactored it so it now takes a string and injects the 'openid' scope if it's not present. I did leave the parameter itself as optional and throw an error if it's not present though. The reason being that it's optional according to the spec if you use the `authorization_details` parameter instead, but we throw because it's currently not supported.